app/core/datastore/src/main/kotlin/com/hedvig/android/core/datastore/SettingsDataStore.kt

```diff
@@ -0,0 +1,44 @@
+package com.hedvig.android.core.datastore
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+interface SettingsDataStore {
+  suspend fun setTheme(theme: String)
+  fun observeTheme(): Flow<String>
+  suspend fun setLanguage(language: String)
+  fun observeLanguage(): Flow<String>
+}
+
+internal class SettingsDataStoreImpl(
+  private val dataStore: DataStore<Preferences>,
+) : SettingsDataStore {
+
+  private val languageKey = stringPreferencesKey("settings-language")
+  private val themeKey = stringPreferencesKey("settings-theme")
+
+  override suspend fun setTheme(theme: String) {
+    dataStore.edit {
+      it[themeKey] = theme
+    }
+  }
+
+  override fun observeTheme(): Flow<String> {
+    return dataStore.data.map { it[themeKey] ?: "LIGHT" }
+  }
+
+  override suspend fun setLanguage(language: String) {
+    dataStore.edit {
+      it[languageKey] = language
+    }
+  }
+
+  override fun observeLanguage(): Flow<String> {
+    return dataStore.data.map { it[languageKey] ?: "EN_SE" }
+  }
```

Comment:
I think for language we should use [LanguageService](https://github.com/HedvigInsurance/android/blob/0d5f2f0a3ca4ca42b18eb98dcabb721648fa3d20/app/language/language-core/src/main/kotlin/com/hedvig/android/language/LanguageService.kt#L15) instead, since that one uses `AppCompatDelegate.getApplicationLocales()` and `AppCompatDelegate.setApplicationLocales` to get and set the language.

The benefit of those APIs is that you can then in compose just use [getLocale()](https://stackoverflow.com/a/75172481/9440211) and you get the latest value automatically. It works with the [per-app language settings](https://developer.android.com/guide/topics/resources/app-languages) that Android 13 and on allow you to edit.

That way we don't need 2 sources of truth for what language the app is in, our settings and Android settings. We just use those APIs and we get 1 place where the truth is in.

From those docs: "These APIs automatically sync with system settings; therefore, apps that use these APIs to create custom in-app language pickers will ensure their users have a consistent user experience regardless of where they select their language preferences".